src/gtasa/effects/custom/player/SwimLikeTommyEffect.cpp

```diff
@@ -0,0 +1,62 @@
+#include "util/EffectBase.h"
+#include "util/GenericUtil.h"
+
+class SwimLikeTommyEffect : public EffectBase
+{
+    int wait = 0;
+
+public:
+    void
+    OnProcessScripts (EffectInstance *inst) override
+    {
+        CPlayerPed *player = FindPlayerPed ();
+        if (player && !FindPlayerVehicle (-1, false))
+        {
+            CPad *pad = player->GetPadFromPlayer ();
+            if (!pad)
+            {
+                return;
+            }
+
+            bool cond = player->m_pIntelligence->GetTaskSwim ();
+            cond |= player->m_nPhysicalFlags.bSubmergedInWater;
+
+            if (cond)

```

Comment:
And same as in the other file, too - Can just do `pad->DisablePlayerControls = cond;` probably